File: samples/SelfAspNet/SelfAspNetApi/Models/SelfAsp.cs
namespace SelfAspNetApi.Models
{
    using System;
    using System.Data.Entity;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Linq;

    public partial class SelfAsp : DbContext
    {
        public SelfAsp()
            : base("name=SelfAspEntities")
        {
        }

        public virtual DbSet<Book> Book { get; set; }

        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Book>()
                .Property(e => e.isbn)
                .IsFixedLength()
                .IsUnicode(false);
        }
    }
}

File: samples/SelfAspNetVb/SelfAspNetApi/SelfAsp.vb
Imports System
Imports System.Data.Entity
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Linq

Partial Public Class SelfAsp
    Inherits DbContext

    Public Sub New()
        MyBase.New("name=SelfAspEntities")
    End Sub

    Public Overridable Property Book As DbSet(Of Book)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Entity(Of Book)() _
            .Property(Function(e) e.isbn) _
            .IsFixedLength() _
            .IsUnicode(False)
    End Sub
End Class
